isisd/isis_adjacency.c

```diff
@@ -93,6 +93,16 @@ struct isis_adjacency *isis_new_adj(const uint8_t *id, const uint8_t *snpa,
 	adj->adj_sids = list_new();
 	listnode_add(circuit->area->adjacency_list, adj);
 
+	/* if BFD is configured on ISIS interface and IPV6 and IPV4 are
+	 * configured start a timer to wait for Zebra to inform ISIS about
+	 * both address families before starting BFD session
+	 */
+	if (circuit->bfd_info && circuit->ipv6_router && circuit->ip_router)
```

Comment:
why do we need a timer?  I don't understand what we are trying to do here.  Can I get a bit of an explanation of how this is supposed to work?

Reply:
@donaldsharp see this comment here (from the issue linked above) to get some context: https://github.com/FRRouting/frr/issues/6386#issuecomment-635293807

Basically if both IPv4 and IPv6 are configured on the circuit there is a race determining which type of BFD session will be created and this commit attempts to work-around it

Reply:
If this analysis is critical then it belongs in the commit message.  This also doesn't answer my question about this approach.  Why do we need a timer?  How does this guarantee that we have all the data?